File: testdata/ocaml/sample.ml
let greet name =
  "Hello, " ^ name ^ "!"

let add x y = x + y

type color = Red | Green | Blue

type point = {
  x: float;
  y: float;
}

module type SHAPE = sig
  val area : point -> float
end

module Circle = struct
  let area r = Float.pi *. r *. r
end

class counter init = object
  val mutable count = init
  method increment = count <- count + 1
  method get = count
end

File: testdata/cpp/sample.cpp
// Greeter class
class Greeter {
public:
    // greet method
    std::string greet(std::string name) {
        return "Hello, " + name + "!";
    }

    // add method
    int add(int a, int b) {
        return a + b;
    }
};

// Point struct
struct Point {
    double x;
    double y;
};

// Color enum
enum class Color { Red, Green, Blue };
